presto-main/src/main/java/com/facebook/presto/operator/aggregation/KeyValuePairs.java

```diff
@@ -39,21 +44,28 @@
     private final BlockBuilder valueBlockBuilder;
     private final Type valueType;
 
+    private boolean isMultiValue;
+
     public KeyValuePairs(Type keyType, Type valueType)
     {
-        checkNotNull(keyType, "keyType is null");
-        checkNotNull(valueType, "valueType is null");
+        this(keyType, valueType, false);
+    }
 
-        this.keyType = keyType;
-        this.valueType = valueType;
+    public KeyValuePairs(Type keyType, Type valueType, Boolean isMultiValue)
+    {
+        this.keyType = checkNotNull(keyType, "keyType is null");
+        this.valueType = checkNotNull(valueType, "valueType is null");
         this.keySet = new SimpleTypedSet(keyType, EXPECTED_ENTRIES);
         keyBlockBuilder = this.keyType.createBlockBuilder(new BlockBuilderStatus(), EXPECTED_ENTRIES, expectedValueSize(keyType, EXPECTED_ENTRY_SIZE));
         valueBlockBuilder = this.valueType.createBlockBuilder(new BlockBuilderStatus(), EXPECTED_ENTRIES, expectedValueSize(valueType, EXPECTED_ENTRY_SIZE));
+
+        this.isMultiValue = checkNotNull(isMultiValue, "isMultiValue is null");
```

Comment:
instead of this null check, just make `isMultiValue` a `boolean` instead of a `Boolean`